vowpalwabbit/json_parser/include/vw/json_parser/parse_example_json.h

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#include "vw/common/future_compat.h"
+#include "vw/core/label_parser.h"
+#include "vw/core/parse_example.h"
+#include "vw/core/parser.h"
+#include "vw/core/v_array.h"
+#include "vw/io/logger.h"
+#include "vw/json_parser/json_utils.h"
+#include "vw/json_parser/parse_example_slates_json.h"
+
+#include <set>
+#include <string>
+#include <unordered_map>
+
+namespace VW
+{
+namespace parsers
+{
+namespace json
+{
+namespace details
+{
+template <bool audit>
+bool parse_line_json(VW::workspace* all, char* line, size_t num_chars, VW::multi_ex& examples);
+}
+
+template <bool audit>
+void read_line_json_s(const VW::label_parser& lbl_parser, hash_func_t hash_func, uint64_t hash_seed,
```

Comment:
What does `_s` mean?

Reply:
I am guessing string, we could rename it to plain `read_line_json`

Reply:
Maybe after this merged we can do a pass cleaning up the functions a bit? We can leave it to later to make this one easier though

Reply:
yes we should, I also wanted to make some char*/size -> string_view transistions but the PR is big already